src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,495 @@
+        val iter: CloseableTribbleIterator[BEDFeature] = bedReader.iterator()
+        val contigToTargets = mutable.HashMap[String, Set[Interval]]()
+        iter.forEach(feature => {
+          val contig = feature.getContig
+          val interval = new Interval(contig, feature.getStart, feature.getEnd)
+          contigToTargets.put(contig, contigToTargets.getOrElse(contig, new HashSet[Interval]()) + interval)
+        })
+        bedReader.close()
+        iter.close()
+        logger.info(f"Read ${contigToTargets.values.map(_.size).sum} targets on ${contigToTargets.keys.size} " +
+          f"contigs from file $bed")
+        Some(contigToTargets.toMap)
+    }
+
+    // Open output writer
+    val writer = Metric.writer[AggregatedBreakpointPileup](output)
+
+    // Group pileups by left and right contig and strand
+    pileups.groupMap(BreakpointContigOrientation(_))(identity).foreach { case (_, group) =>
+
+      // Within each contig+strand group, identify all immediate "neighbors" of each pileup (other pileups within the
+      // distance threshold)
+      val pileupToNeighbors: Map[BreakpointPileup, PileupGroup] = AggregateSvPileup.pileupToNeighbors(group, maxDist)
+
+      // Within each contig+strand group, recursively build (transitive) clusters of neighboring pileups
+      val pileupClusters: Seq[PileupGroup] = AggregateSvPileup.toClusters(pileupToNeighbors)
+      logger.info(f"Processing ${pileupClusters.size} pileup clusters with endpoints " +
+        f"${group.head.left_contig}:${group.head.left_strand}, ${group.head.right_contig}:${group.head.right_strand}")
+
+      // Aggregate each cluster, analyze allele frequency and target overlap, and write output
+      pileupClusters
+        .map(AggregatedBreakpointPileup.apply)
+        .map(_.calculateFrequency(samSource, flank, minTotal, minFrequency))
+        .map(_.addTargetOverlap(targets))
+        .foreach(writer.write)
+    }
+
+    writer.close()
+
+  }
+
+}
+
+
+object AggregateSvPileup {
+
+  /** Type alias for a collection of pileups */
+  type PileupGroup = Seq[BreakpointPileup]
+
+  /**
+   * Maps each pileup to the set of other pileups within the distance threshold.
+   *
+   * Note: it is recommended to call this method on a group of pileups that share a common `BreakpointContigOrientation`
+   * to reduce unnecessary run time.
+   *
+   * @param pileups Set of pileups to evaluate for pairwise neighbor relationships
+   * @param maxDist Maximum distance (inclusive) for the left and right coordinates to be considered a neighbor
+   * @return Map of pileup to the set of neighbors
+   */
+  private def pileupToNeighbors(pileups: PileupGroup, maxDist: Int): Map[BreakpointPileup, PileupGroup] = {
+    val map: mutable.Map[BreakpointPileup, List[BreakpointPileup]] = mutable.Map.from(pileups.map((_, Nil)))
+    for (pileup1 <- pileups; pileup2 <- pileups) {
+      if (pileup1.total >= pileup2.total && isCluster(pileup1, pileup2, maxDist)) {
```

Comment:
Is it intentional to include `pileup1` in it's own pileup group?

Reply:
I still think it would be better to:
1. When you read in the metrics above do a `toIndexedSeq`
2. Redefine PileupGroup to `IndexedSeq[BreakpointPileup]`
3. Redo the loop below to something like:

```scala
for (i <- pileups.indices; j <- pileups.indices.drop(i) {
  val pileup1 = pileups(i)
  val pileup2 = pileups(j)
  ...
}
```

It significantly reduces the looping and eliminates the need for the `.distinct` calls I believe.